mslice/app/__init__.py

```diff
@@ -25,4 +25,5 @@ def show_gui():
     if MAIN_WINDOW is None:
         from mslice.app.mainwindow import MainWindow
         MAIN_WINDOW = MainWindow()
+        MAIN_WINDOW.setup_ipython()
```

Comment:
Can you help me understand why we don't do this as part of the constructor anymore?

Reply:
I can't remember. I changed it back and everything seems fine so I guess I don't understand either.

Reply:
I remember why I did this now  - if `setup_ipython` is done as part of the `MainWindow` constructor, the CLI commands can't find the `MainWindow` instance e.g.

```
AttributeError                            Traceback (most recent call last)
<ipython-input-3-8acdd31d31dc> in <module>()
----> 1 ws = Load('C:\Users\ygh95688\Documents\MSlice_Test_Data\MAR21335_ei60.00meV.nxs')

C:\Users\ygh95688\PycharmProjects\mslice_2\mslice\cli\_mslice_commmands.pyc in Load(path)
     84     if not ospath.exists(path):
     85         raise RuntimeError('could not find the path %s' % path)
---> 86     MAIN_WINDOW.dataloader_presenter.load_workspace([path])
     87     return get_workspace_handle(ospath.splitext(ospath.basename(path))[0])
     88 

AttributeError: 'NoneType' object has no attribute 'dataloader_presenter'
```

Is there another way around this other than moving the `setup_ipython` call?

Reply:
It's not clear to me why `MAIN_WINDOW` has not been set. The only way I can see that `MAIN_WINDOW` is `None` is if the `MainWindow` constructor raises an exception. Is that the case?

Reply:
It's not clear to me either. I stepped through the constructor in the debugger and everything seems to run fine.

Reply:
This was a problem with the way `MAIN_WINDOW` was imported meaning it wasn't updated, should be ok now.

Reply:
Ah yes, I understand now. :+1: